Sources/TuistKit/Services/InitService.swift

```diff
@@ -92,23 +98,42 @@ class InitService {
 
         let directories = try templatesDirectoryLocator.templateDirectories(at: path)
         if let templateName = templateName {
-            guard
-                let templateDirectory = directories.first(where: { $0.basename == templateName })
-            else { throw InitServiceError.templateNotFound(templateName) }
-            let template = try templateLoader.loadTemplate(at: templateDirectory)
-            let parsedAttributes = try parseAttributes(
-                name: name,
-                platform: platform,
-                requiredTemplateOptions: requiredTemplateOptions,
-                optionalTemplateOptions: optionalTemplateOptions,
-                template: template
-            )
+            var template: Template?
+            if templateName.isURL {
+                try fileHandler.inTemporaryDirectory { temporaryDirectory in
+                    let templateDirectoryName = temporaryDirectory.appending(.init("\(templateName)")).url.lastPathComponent
+                    let templateDirectory = temporaryDirectory.appending(.init("\(templateDirectoryName)"))
+                    try fileHandler.createFolder(templateDirectory)
+                    try gitHandler.clone(url: templateName, to: templateDirectory)
+                    template = try templateLoader.loadTemplate(at: templateDirectory)
+                    guard let template = template else { return }
```

Comment:
We should throw an error here, otherwise users won't know why the project was not initialized.